js/main.js

```diff
@@ -21,15 +26,18 @@ require('./memoryMonitor.js');
 const windowMgr = require('./windowMgr.js');
 
 // only allow a single instance of app.
-const shouldQuit = app.makeSingleInstance(() => {
+const shouldQuit = app.makeSingleInstance((argv) => {
     // Someone tried to run a second instance, we should focus our window.
     let mainWin = windowMgr.getMainWindow();
     if (mainWin) {
+        isAppAlreadyOpen = true;
         if (mainWin.isMinimized()) {
             mainWin.restore();
         }
         mainWin.focus();
     }
+    // Handle protocol action if necessary
+    processProtocolAction(argv);
```

Comment:
not exactly. `app.singleInstance` is called when we already have the app running, that means, we'll have the main window initialized in which the control will not go into the `if` condition. So, the `processProtocolAction` should be where it is now.